src/Tasks/Microsoft.NET.Build.Tasks/ProjectContext.cs

```diff
@@ -56,10 +57,22 @@ public IEnumerable<LockFileTargetLibrary> GetRuntimeLibraries(IEnumerable<string
                 allExclusionList.UnionWith(privateAssetsExclusionList);
             }
 
-            if(_filterlockFileTarget != null)
+            if(_packagesToBeFiltered != null)
             {
-                IEnumerable<LockFileTargetLibrary> filterLibraries = _filterlockFileTarget.Libraries;
-                Dictionary<string, LockFileTargetLibrary> filterLookup = filterLibraries.ToDictionary(e => e.Name, StringComparer.OrdinalIgnoreCase);
+                var filterLookup = new Dictionary<string, HashSet<PackageInfo>>();
+                foreach (var pkg in _packagesToBeFiltered)
+                {
+                    if (filterLookup.ContainsKey(pkg.Name))
+                    {
+                        filterLookup[pkg.Name].Add(pkg);
```

Comment:
Why are we trying to add a new value for an existing key? Per https://msdn.microsoft.com/en-us/library/k7z0zy8k(v=vs.110).aspx, Add would throw an exception.

Instead, should the key be made unique so that such a conflict does not happen when there are multiple entries for different versions of a given package?